.github/workflows/release.yml

```diff
@@ -0,0 +1,58 @@
+name: Release
+
+on:
+  workflow_dispatch:
+    inputs:
+      versionType:
+        description: "Version Type - Major, Minor, Patch, Manual"
+        required: true
+        default: "Patch"
+        type: choice
+        options:
+          - major
+          - minor
+          - patch
+          - manual
+      customVersion:
+        description: "Custom Version - Use if Version Type is Manual"
+        required: false
+
+jobs:
+  release:
+    # validate if input was versionType = Major, Minor, Patch or Manual
+    # if its Manual we require the `customVersion` value
+    if: github.event.inputs.versionType != 'manual' || github.event.inputs.customVersion != null
+
+    runs-on: ubuntu-latest
+    steps:
+      - name: Checkout code
+        uses: actions/checkout@v4
+        with:
+          fetch-depth: 100
```

Comment:
Curious if we can/should reduce this. Would we be able to get away with 1 here?

Reply:
I did a little research and I think the best we can get away with is by using `filter: tree:0`:
https://github.com/actions/checkout/issues/1152#issuecomment-1752251377

The reason is that we need the history to be able to figure out the latest git tag. A shallow clone unfortunately does not give us that.

Maybe we can reduce the depth to 50, or 25 assuming that we will always cut new releases when we are about 50 or 25 merged PRs. WDYT?

Reply:
Gotcha! I think it's fine to leave at 100 then, as that should usually contain the last tag. Probably a bit better to overestimate than under :)